### DISCUSSION TITLE
how to use my own yolov5 model to infer

### DISCUSSION THREAD

**Discussion by 66Ethanwang** [obss/sahi#669] [Q&A] [answered]
Excuse me，i trained my own yolov5 model and i already added opts in to the commen.py ,but It can't read the correct parameters of the backbone，I don't know if I overlooked something。

Below is my backbone parameter：
yolov5.models.yolo -     0                -1  1      3520  yolov5.models.common.Conv               [3, 32, 6, 2, 2]              
 yolov5.models.yolo -    1                -1  1      1932  yolov5.models.common.RepVGGBlock        [128, 3, 2]                   
yolov5.models.yolo -     2                -1  1     16768  yolov5.models.common.C3                 [32, 64, 1]                   
yolov5.models.yolo -     3                -1  1      3852  yolov5.models.common.RepVGGBlock        [256, 3, 2]

**Comment by kadirnar** [comment]
Hi @[66Ethanwang](https://github.com/66Ethanwang) 

To test the custom Yolov5 model you should follow the steps here.
https://github.com/kadirnar/yolov5-sahi

**Comment by 66Ethanwang** [comment]
got it ,and thanks  a lot.
I finally found the reason for this problem because I modified the operator in the file common.py

**Comment by fcakyon** [comment, marked as answer]
@66Ethanwang yes custom implementations of yolov5 are not supported on the fly, you have to update sahi/models.py for your needs 👍

**Comment by 66Ethanwang** [comment]
got it,but i have another question , in the function get_sliced_prediction ,i set the  slice_height=1680,and slice_width=1680, After image slicing，Yolov5 does it directly use (1680，1680) to predict? Or where it will use the letter_box? i debug many times but I couldn't find the answer.

**Comment by fcakyon** [comment]
@66Ethanwang by default yolov5 will perform inference on 640 size, you can update `image_size` param at Yolov5DetectionModel class init.

**Comment by gubei528** [comment]
https://github.com/kadirnar/yolov5-pytorch-sahi/blob/main/yolov5_sahi.py This website can't be opened! Does my own yolo model support it?

**Reply by kadirnar** [reply]
Hello, this is sahi-yolov5 guide for turkish users.
- https://github.com/kadirnar/yolov5-sahi

**Reply by gubei528** [reply]
thank you for your reply

**Comment by Hehezcr** [comment]
请问您是如何将sahi跟自己改过的网络结构的YOLOV5结合的，我已经按照要求在model文件夹中创建了一个myyolov5.py文件，并且模仿着写了一个Yolov5HubDetectionModel，但是一运行就报ValueError: Model is not loaded, load it by calling .load_model()的错误，应该如何解决
